filip/models/ngsi_v2/context.py

```diff
@@ -203,49 +203,173 @@ def _validate_properties(cls, data: Dict):
+        Update attributes of an entity. Overwrite the current held value
+        for the attribute with the value contained in the corresponding given
+        attribute
+
         Args:
-            attrs:
+            attrs: List of NamedContextAttributes,
+                   Dict of {attribute_name: ContextAttribute}
+        Raises:
+            NameError, if the attribute does not currently exists in the entity
         Returns:
             None
         """
         if isinstance(attrs, list):
             attrs = {attr.name: ContextAttribute(**attr.dict(exclude={'name'}))
                      for attr in attrs}
+
+        existing_attribute_names = self.get_attribute_names()
         for key, attr in attrs.items():
+            if key not in existing_attribute_names:
+                raise NameError
             self.__setattr__(name=key, value=attr)
 
-    def get_relationships(self,
-                          response_format: Union[str, PropertyFormat] =
-                          PropertyFormat.LIST) \
-            -> Union[List[NamedContextAttribute],
-                     Dict[str, ContextAttribute]]:
+    def get_attribute_names(self) -> Set[str]:
+        """
+        Returns a set with all attribute names of this entity
+
+        Returns:
+            Set[str]
+        """
+
+        return {key for key in self.dict()
+                if key not in ContextEntity.__fields__}
+
+    def delete_attributes(self, attrs: Union[Dict[str, ContextAttribute],
+                                             List[NamedContextAttribute],
+                                             List[str]]):
+        """
+        Delete the given attributes from the entity
+
+        Args:
+            attrs:  - Dict {name: ContextAttribute}
+                    - List[NamedContextAttribute]
+                    - List[str] -> names of attributes
+        Raises:
+            Exception: if one of the given attrs does not represent an
+                       existing argument
+        """
+
+        names: List[str] = []
+        if isinstance(attrs, list):
+            for entry in attrs:
+                if isinstance(entry, str):
+                    names.append(entry)
+                elif isinstance(entry, NamedContextAttribute):
+                    names.append(entry.name)
+        else:
+            names.extend(list(attrs.keys()))
+        for name in names:
+            delattr(self, name)
+
+    def get_attribute(self, attribute_name: str) -> NamedContextAttribute:
+        """
+        Get the attribute of the entity with the given name
+
+        Args:
+            attribute_name (str): Name of attribute
+        Raises:
+            KeyError, if no attribute with given name exists
+        Returns:
+            NamedContextAttribute
+
+        """
+        for attr in self.get_attributes():
+            if attr.name == attribute_name:
+                return attr
+        raise KeyError
+
+    def get_properties(
```

Comment:
why not also a function for adding a property? to similar to add_attribute which could also add a relationship?

Reply:
For the naming I oriented myself on the existing ideas. Each (Named)ContextAttribute is an attribute, if it further has the type RELATIONSHIP it is also a relationship else a property. 
As it is not important for the adding process if you add a property or relationship all 'add' functions are kept in "superclass style" (attribute), same with update and delete. 

Only the getting functions are interesting for the user to directly get Relationships or all properties (as it was implemented before), therefore only at that place the additional filtering functions exist.